File: backend/api/DTO/StudySessionCreateRequest.cs
namespace api.DTO;

public class StudySessionCreateRequest
{
    public int UserId { get; set; }
    public int DeckId { get; set; }
    public int Duration { get; set; }
    public int CardCount { get; set; }
    public float AverageEaseFactor { get; set; }
    public float AverageRepetitions { get; set; }
    public DateTime StudiedAt { get; set; }
    public int CardsRated1 { get; set; }
    public int CardsRated2 { get; set; }
    public int CardsRated3 { get; set; }
    public int CardsRated4 { get; set; }
    public int CardsRated5 { get; set; }
}

File: models/study-session-create-request.ts
export class StudySessionCreateRequest {
  deck_id: number;
  duration: number;
  correct_answers: number;
  incorrect_answers: number;

  constructor({
    deck_id,
    duration,
    correct_answers,
    incorrect_answers,
  }: {
    deck_id: number;
    duration: number;
    correct_answers: number;
    incorrect_answers: number;
  }) {
    this.deck_id = deck_id;
    this.duration = duration;
    this.correct_answers = correct_answers;
    this.incorrect_answers = incorrect_answers;
  }
}
